lib/accountregistry.h

```diff
@@ -0,0 +1,58 @@
+// SPDX-FileCopyrightText: 2020 Kitsune Ral <Kitsune-Ral@users.sf.net>
+// SPDX-FileCopyrightText: Tobias Fella <fella@posteo.de>
+// SPDX-License-Identifier: LGPL-2.1-or-later
+
+#pragma once
+
+#include <QtCore/QObject>
+#include <QtCore/QList>
+#include <QtCore/QAbstractListModel>
+
+namespace Quotient {
+class Connection;
+
+class AccountRegistry : public QAbstractListModel {
+    Q_OBJECT
+public:
+    using const_iterator = QMap<QString, Connection *>::const_iterator;
+
+    enum EventRoles {
+        ConnectionRole = Qt::UserRole + 1,
+    };
+
+    static AccountRegistry &instance() {
```

Comment:
`Q_GLOBAL_STATIC`? Though I'm not totally sure how much one is better than another in this case.